### ISSUE TITLE
Visual studio 2019 errors include format library

### ISSUE THREAD

**Issue by Widualk** [fmtlib/fmt#1211]
Hi , i create new project, i try to load library fmt , i modify Properties > C/C++ > General > Additional Include Directory > fmt-5.3.0> include > fmt.

I get this errors:
`1>------ Build started: Project: ConsoleApplication1, Configuration: Debug Win32 ------
1>pch.cpp
1>ConsoleApplication1.cpp
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocbuf(215,50): error C2039:  'seekdir': is not a member of 'std::ios_base'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\iosfwd(163): message :  see declaration of 'std::ios_base'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocbuf(290): message :  see reference to class template instantiation 'std::wbuffer_convert<_Codecvt,_Elem,_Traits>' being compiled
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocbuf(215,1): error C2061:  syntax error: identifier 'seekdir'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocbuf(221,19): error C2039:  'openmode': is not a member of 'std::ios_base'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\iosfwd(163): message :  see declaration of 'std::ios_base'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocbuf(221,1): error C2061:  syntax error: identifier 'openmode'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(17,39): error C2027:  use of undefined type 'std::locale'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\iosfwd(142): message :  see declaration of 'std::locale'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(17,45): error C2504:  'facet': base class undefined
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(20,56): error C2027:  use of undefined type 'std::locale'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\iosfwd(142): message :  see declaration of 'std::locale'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(20,46): error C2614:  'std::messages_base': illegal member initialization: 'facet' is not a base or member
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(47,65): error C2039:  'id': is not a member of 'std::locale'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\iosfwd(142): message :  see declaration of 'std::locale'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(92): message :  see reference to class template instantiation 'std::messages<_Elem>' being compiled
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(47,1): error C3646:  'id': unknown override specifier
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(47,1): error C4430:  missing type specifier - int assumed. Note: C++ does not support default-int
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(55,1): error C4430:  missing type specifier - int assumed. Note: C++ does not support default-int
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(55,28): error C2143:  syntax error: missing ',' before '&'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(59,41): error C2039:  'facet': is not a member of 'std::locale'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\iosfwd(142): message :  see declaration of 'std::locale'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(59,1): error C4430:  missing type specifier - int assumed. Note: C++ does not support default-int
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(59,46): error C2143:  syntax error: missing ',' before '*'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(78,1): error C4430:  missing type specifier - int assumed. Note: C++ does not support default-int
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(78,30): error C2143:  syntax error: missing ',' before '&'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(50,9): error C2760:  syntax error: unexpected token 'identifier', expected ';'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(55,71): error C3861:  '_Refs': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(56,15): error C3861:  '_Lobj': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(61,13): error C3861:  '_Ppf': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(61,33): error C3861:  '_Ppf': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(62,14): error C3861:  '_Ppf': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(62,50): error C3861:  '_Ploc': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(65,16): error C3861:  '_X_MESSAGES': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(70,9): error C2760:  syntax error: unexpected token 'identifier', expected ';'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(101,33): error C2039:  'id': is not a member of 'std::locale'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\iosfwd(142): message :  see declaration of 'std::locale'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(101,53): error C2143:  syntax error: missing ';' before 'std::messages<_Elem>::id'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(101,1): warning C4346:  'id': dependent name is not a type
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(101,1): message :  prefix with 'typename' to indicate a type
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(129,42): error C2039:  'id': is not a member of 'std::locale'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\iosfwd(142): message :  see declaration of 'std::locale'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(47,65): error C2027:  use of undefined type 'std::locale'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\iosfwd(142): message :  see declaration of 'std::locale'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(129): message :  see reference to class template instantiation 'std::messages<char>' being compiled
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(59,41): error C2027:  use of undefined type 'std::locale'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\iosfwd(142): message :  see declaration of 'std::locale'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(129,61): error C2143:  syntax error: missing ';' before 'std::messages<char>::id'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(130,42): error C2039:  'id': is not a member of 'std::locale'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\iosfwd(142): message :  see declaration of 'std::locale'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmes(130,64): error C2143:  syntax error: missing ';' before 'std::messages<wchar_t>::id'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(17,36): error C2039:  'facet': is not a member of 'std::locale'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\iosfwd(142): message :  see declaration of 'std::locale'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(17,42): error C2504:  'facet': base class undefined
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(31,34): error C2059:  syntax error: ':'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(88,1): error C4430:  missing type specifier - int assumed. Note: C++ does not support default-int
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(251): message :  see reference to class template instantiation 'std::_Mpunct<_Elem>' being compiled
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(88,27): error C2143:  syntax error: missing ',' before '&'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(106,1): error C4430:  missing type specifier - int assumed. Note: C++ does not support default-int
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(106,38): error C2988:  unrecognizable template declaration/definition
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(106,38): error C2143:  syntax error: missing ',' before '*'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(117,1): error C4430:  missing type specifier - int assumed. Note: C++ does not support default-int
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(117,39): error C2143:  syntax error: missing ',' before '*'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(125,1): error C4430:  missing type specifier - int assumed. Note: C++ does not support default-int
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(125,30): error C2143:  syntax error: missing ',' before '&'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(250,1): error C2653:  '_Locinfo': is not a class or namespace name
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(250,1): error C3646:  '_Cvt': unknown override specifier
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(250,1): error C4430:  missing type specifier - int assumed. Note: C++ does not support default-int
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(83,9): error C2760:  syntax error: unexpected token 'identifier', expected ';'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(89,22): error C3861:  '_Refs': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(89,2): error C3861:  '_Intl': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(90,15): error C3861:  '_Lobj': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(90,22): error C3861:  '_Isdef': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(96,9): error C2760:  syntax error: unexpected token 'identifier', expected ';'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(108,30): error C3861:  '_Ptr': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(108,30): message :  This diagnostic occurred in the compiler generated function 'void std::_Mpunct<_Elem>::_Getvals(_Elem2,const int)'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(108,54): error C3861:  '_Ptr': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(108,54): message :  This diagnostic occurred in the compiler generated function 'void std::_Mpunct<_Elem>::_Getvals(_Elem2,const int)'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(108,108): error C3861:  '_Cvt': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(108,108): message :  This diagnostic occurred in the compiler generated function 'void std::_Mpunct<_Elem>::_Getvals(_Elem2,const int)'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(110,32): error C3861:  '_Ptr': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(110,32): message :  This diagnostic occurred in the compiler generated function 'void std::_Mpunct<_Elem>::_Getvals(_Elem2,const int)'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(110,57): error C3861:  '_Ptr': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(110,57): message :  This diagnostic occurred in the compiler generated function 'void std::_Mpunct<_Elem>::_Getvals(_Elem2,const int)'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(110,109): error C3861:  '_Cvt': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(110,109): message :  This diagnostic occurred in the compiler generated function 'void std::_Mpunct<_Elem>::_Getvals(_Elem2,const int)'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(112,32): error C3861:  '_Ptr': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(112,32): message :  This diagnostic occurred in the compiler generated function 'void std::_Mpunct<_Elem>::_Getvals(_Elem2,const int)'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(112,58): error C3861:  '_Ptr': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(112,58): message :  This diagnostic occurred in the compiler generated function 'void std::_Mpunct<_Elem>::_Getvals(_Elem2,const int)'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(112,110): error C3861:  '_Cvt': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(112,110): message :  This diagnostic occurred in the compiler generated function 'void std::_Mpunct<_Elem>::_Getvals(_Elem2,const int)'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(113,36): error C3861:  '_Ptr': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(113,36): message :  This diagnostic occurred in the compiler generated function 'void std::_Mpunct<_Elem>::_Getvals(_Elem2,const int)'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(113,95): error C3861:  '_Cvt': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(113,95): message :  This diagnostic occurred in the compiler generated function 'void std::_Mpunct<_Elem>::_Getvals(_Elem2,const int)'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(114,36): error C3861:  '_Ptr': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(114,36): message :  This diagnostic occurred in the compiler generated function 'void std::_Mpunct<_Elem>::_Getvals(_Elem2,const int)'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(114,95): error C3861:  '_Cvt': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(114,95): message :  This diagnostic occurred in the compiler generated function 'void std::_Mpunct<_Elem>::_Getvals(_Elem2,const int)'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(118,68): error C3861:  '_Ptr': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(118,95): error C3861:  '_Ptr': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(119,70): error C3861:  '_Ptr': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(119,96): error C3861:  '_Ptr': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(120,70): error C3861:  '_Ptr': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(120,97): error C3861:  '_Ptr': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(121,33): error C3861:  '_Ptr': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(122,33): error C3861:  '_Ptr': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(126,9): error C3861:  '_Cvt': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(126,29): error C3861:  '_Lobj': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(127,15): error C3861:  'lconv': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(127,29): error C3861:  '_Lobj': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(135,81): error C3861:  '_Cvt': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(149,13): error C3861:  '_Isdef': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(262,65): error C2039:  'id': is not a member of 'std::locale'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\iosfwd(142): message :  see declaration of 'std::locale'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(287): message :  see reference to class template instantiation 'std::moneypunct<_Elem,_Intl>' being compiled
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(262,1): error C3646:  'id': unknown override specifier
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(262,1): error C4430:  missing type specifier - int assumed. Note: C++ does not support default-int
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(267,1): error C4430:  missing type specifier - int assumed. Note: C++ does not support default-int
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(267,30): error C2143:  syntax error: missing ',' before '&'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(271,41): error C2039:  'facet': is not a member of 'std::locale'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\iosfwd(142): message :  see declaration of 'std::locale'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(271,1): error C4430:  missing type specifier - int assumed. Note: C++ does not support default-int
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(271,46): error C2143:  syntax error: missing ',' before '*'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(268,26): error C3861:  '_Lobj': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(268,33): error C3861:  '_Refs': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(268,47): error C3861:  '_Isdef': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(273,13): error C3861:  '_Ppf': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(273,33): error C3861:  '_Ppf': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(274,14): error C3861:  '_Ppf': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(274,12): error C3861:  '_Ploc': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(277,16): error C3861:  '_X_MONETARY': identifier not found
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(300,33): error C2039:  'id': is not a member of 'std::locale'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\iosfwd(142): message :  see declaration of 'std::locale'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(300,62): error C2143:  syntax error: missing ';' before 'std::moneypunct<_Elem,_Intl>::id'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(300,1): warning C4346:  'id': dependent name is not a type
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(300,1): message :  prefix with 'typename' to indicate a type
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(327,34): error C2039:  'facet': is not a member of 'std::locale'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\iosfwd(142): message :  see declaration of 'std::locale'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(614): message :  see reference to class template instantiation 'std::money_get<_Elem,_InIt>' being compiled
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(339,84): error C2039:  'iostate': is not a member of 'std::ios_base'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\iosfwd(163): message :  see declaration of 'std::ios_base'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(339,1): error C2061:  syntax error: identifier 'iostate'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(344,84): error C2039:  'iostate': is not a member of 'std::ios_base'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\iosfwd(163): message :  see declaration of 'std::ios_base'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(344,1): error C2061:  syntax error: identifier 'iostate'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(345,1): error C2535:  '_InIt std::money_get<_Elem,_InIt>::get(_InIt,_InIt,bool,std::ios_base &) const': member function already defined or declared
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(339): message :  see declaration of 'std::money_get<_Elem,_InIt>::get'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(349,65): error C2039:  'id': is not a member of 'std::locale'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\iosfwd(142): message :  see declaration of 'std::locale'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(349,1): error C3646:  'id': unknown override specifier
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(349,1): error C4430:  missing type specifier - int assumed. Note: C++ does not support default-int
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(357,1): error C4430:  missing type specifier - int assumed. Note: C++ does not support default-int
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(357,29): error C2143:  syntax error: missing ',' before '&'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(361,41): error C2039:  'facet': is not a member of 'std::locale'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\iosfwd(142): message :  see declaration of 'std::locale'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(361,1): error C4430:  missing type specifier - int assumed. Note: C++ does not support default-int
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(361,46): error C2143:  syntax error: missing ',' before '*'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(374,1): error C4430:  missing type specifier - int assumed. Note: C++ does not support default-int
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(374,30): error C2143:  syntax error: missing ',' before '&'
1>D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocmon(374,1): error C1003:  error count exceeds 100; stopping compilation
1>Done building project "ConsoleApplication1.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
`

**Comment by vitaut** [comment]
Have you tried the master branch? It should work with msvc2019.

**Comment by Widualk** [comment]
Give me some steps please

**Comment by Widualk** [comment]
ok it's a simple code:
`#include <stdio.h>
#include <format.h>

int main()
{
	return 0;
}`
and errors from up.
How can i set Visual Studio Comunity to use fmtlib? What setting i need to do?

**Comment by vitaut** [comment]
Just add the header and source files to your project or use CMake. See https://fmt.dev/latest/usage.html

**Comment by Widualk** [comment]
Ok, i add headers and source files, but i get same errors.

**Comment by vitaut** [comment]
Since {fmt} doesn't use `seekdir`, the error
```
D:\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.21.27702\include\xlocbuf(215,50): error C2039: 'seekdir': is not a member of 'std::ios_base'
```
suggests that something is wrong in your build config.

Here's an example of compiling {fmt} with MSVC2019 on godbolt in case it helps: https://godbolt.org/z/2nixbr

**Comment by Widualk** [comment]
Ok i try to add on visual studio include directories same like this: $(SolutionDir)\Project1
my code is:
`#include <stdio.h>
#include "fmt/format.h"
int main()
{
	//fmt::print("Hello, {}!", "world");
	return 0;
}`
it's execute with succes.
and when i try:

`#include <stdio.h>
#include "fmt/format.h"
int main()
{
	fmt::print("Hello, {}!", "world");
	return 0;
}`

i get:
`1>------ Build started: Project: Project1, Configuration: Debug Win32 ------
1>Source.cpp
1>Source.obj : error LNK2019: unresolved external symbol "void __cdecl fmt::v5::vprint(class fmt::v5::basic_string_view<char>,struct fmt::v5::format_args)" (?vprint@v5@fmt@@YAXV?$basic_string_view@D@12@Uformat_args@12@@Z) referenced in function "void __cdecl fmt::v5::print<char [11],char [6]>(char const (&)[11],char const (&)[6])" (??$print@$$BY0L@D$$BY05D@v5@fmt@@YAXAAY0L@$$CBDAAY05$$CBD@Z)
1>C:\Users\Claudiu\source\repos\Project1\Debug\Project1.exe : fatal error LNK1120: 1 unresolved externals
1>Done building project "Project1.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
`

It's a problem with add includes method.
I want to specify in my C/C++ > GENERAL > Additional include directories the headers, and i build i get errors.
When i include : #include <format.h> its recognized the format header, but i get error. 

Problem is with method to add librarys. What method to use?

What

**Comment by Widualk** [comment]
Have you a discord account or something, to have a conversation live?

**Comment by vitaut** [comment]
I don't have Discord but there is a gitter chat: https://gitter.im/fmtlib/fmt . I won't be able to help with MSVC setup since I don't have a Windows machine at hand, but someone else might.

**Comment by Widualk** [comment]
I send you message

**Comment by vlad-mal** [comment]
Please tell us what caused this error? I have the same one now.

**Comment by code-affinity** [comment]
Both issues about this problem were closed without explanation.   (#2071 is the other issue about the same errors.)  This thread is the top Google search hit for "xlocbuf seekdir", which is what I searched for when I encountered the error messages reported in this issue.

Here is what I found in my case:  My copy of the fmt library was built with one version of Visual Studio 2019.  Later, I updated Visual Studio 2019.  Then I built my application and got all of these errors, starting with " 'seekdir': is not a member of 'std::ios_base' ".  I rebuilt fmt with the upgraded Visual Studio 2019.  Then I rebuilt my application.  The compiler errors went away.
